invenio_accounts/api.py

```diff
@@ -8,25 +8,68 @@
 
 """API objects for Invenio Accounts."""
 
-from collections import defaultdict
+
+class Session:
+    """Session object for DB Users change history."""
+
+    def __init__(self):
+        """Constructor."""
+        self.updated_users = set()
+        self.updated_roles = set()
+        self.deleted_users = set()
+        self.deleted_roles = set()
+        self.indexed = False
+        self.invalidated_cache = False
+
+    def index(self):
```

Comment:
as discussed, remove these 2 methods, and add a `finalizer` to invenio-accounts/datastore -> commit method, after the signals.